tests/cli/test_command_list.py

```diff
@@ -141,11 +141,20 @@ def test_final_results_get_no_results(mocker: MockerFixture) -> None:
     mock_remote_backend_inst.get_final_results.assert_called_once()
 
 
-def test_login(mocker: MockerFixture, mocked_config_file: MagicMock) -> None:
+@pytest.mark.parametrize("use_local_auth_config", [True, False])
+def test_login(mocker: MockerFixture, mocked_config_file: MagicMock, use_local_auth_config: bool) -> None:
     device_session = mocker.patch("quantuminspire.cli.command_list.OauthDeviceSession")()
     webbrowser_open = mocker.patch("quantuminspire.cli.command_list.webbrowser.open")
     store_tokens = mocker.patch("quantuminspire.cli.command_list.Settings.store_tokens")
-    result = runner.invoke(app, ["login", "https://host"])
+    fetch_auth = mocker.patch("quantuminspire.cli.command_list.Settings.fetch_suggested_auth_settings")
+
+    if use_local_auth_config:
+        result = runner.invoke(app, ["login", "https://host", "--use-local-auth-config"])
+        fetch_auth.assert_not_called()
+    else:
+        result = runner.invoke(app, ["login", "https://host"])
+        fetch_auth.assert_called_once()
```

Comment:
It feels weird to have a parameterized test, where you need an if-statement to test both behaviours. I also don't think we should include the CLI behaviour on the flag in a test whether login succeeds. With or without the flag, you still open the browser. You just do a request to an endpoint, or read a local file.

Reply:
Yes but to reach full coverage we have to cover both cases:
```
    if not override_auth_config:
        asyncio.run(settings.fetch_auth_settings(host_url))
 ```
 
 Thats why I included both cases here. Reason I decided for one paremterized test is splitting them in two would end up with mostly duplicate code

Reply:
Actually I think I get your point now, have a seperate test for testing the `fetch_auth` method, and leave the `login` without including this logic. I will make the change